machinery/types.go

```diff
@@ -15,10 +15,10 @@ type Object interface {
 
 	GetNamespace() string
 	GetName() string
-	GetURL() string
+	GetIdentity() string
```

Comment:
An alternative that might suit also is `GetRef()/GetReference()` that aligns with the `targetRef` concept 🤔

Reply:
I thought about something along the lines of "reference" too. Maybe `ReferentId` or `ReferentIdentifier` or yet `ReferentUniqueId`, `ReferentUID`, etc. It makes sense for objects that are meant to be linked to others in the topology (from a `targetRef` or from any kind of reference really).

But truth is this is about uniquely identifying the objects before being about linking them. I suppose a more generic term that captures this idea, that is extensible to unlinked objects too, is preferable. It bothers me though that "UID" is taken and we cannot use it 😕

"Identity" is good. I don't love it though. IDK why. Maybe because it's too close to UID and we don't want people to get confused about these two.

Other options that crossed to mind include `Handle` and `Locator`.

Either "identity", "handle" or "locator" (or anything along these lines really), another open question I have regards to qualifying the term:
- "unique", to explicit this constraint
- "topology", to denote purpose – though arguably it can be used to other purposes too
- "descriptive" or "semantic", in contrast to "opaque" (like Kube's UID)

So, any combination of these could work. E.g. "TopologyUniqueId", "DescriptiveHandle", "UniqueSemanticLocator" 🤯

Anyway... Right now I'm kinda leaning to "locator". But I also don't know why. WDYT?

Reply:
Hmm, yeah `Identity` sounds closely related to the kube `UID` 🤔  I like `Locator` or `UniqueLocator` if we are choosing a combination 👍 

@Boomatang Do you have any thoughts on this?

Reply:
I don't have that strong of an opinion on the naming. The term to `Locator` sounds good, it gives a good indication of what the resulting function return should be used for.

Reply:
Updated `GetUrl()` to `GetLocator()`